### PR TITLE
21613-DebugActions-should-have-meaningful-tooltips

### PR DESCRIPTION
**Pull request by jecisc**
Case 21613 : https://pharo.fogbugz.com/f/cases/21613/DebugActions-should-have-meaningful-tooltips

- Remove DebugAction>>defaultHelp which was really not useful...
- Make DebugAction>>help as subclassResponsibility
- Implement #help on all subclasses of DebugAction missing it
- Categorise some uncategorised methods

Working on this I also saw that the context menus do not display tooltips on hover :( This is another issue. (https://pharo.fogbugz.com/f/cases/21614/Debugger-menus-do-not-show-tooltips-of-the-debug-actions)

### CODE REVIEW

File: src/DebuggerActions/FullStackDebugAction.class.st [hunk 1]
```diff
@@ -45,6 +45,11 @@ FullStackDebugAction >> expandedStackSize [
 	^ 100000
 ]
 
+{ #category : #accessing }
+FullStackDebugAction >> help [
+	^ 'Open a full debuggeur to navigate in the full bugged stack.'
```
Comment (iNode) [inline src/DebuggerActions/FullStackDebugAction.class.st]: "debuggeur" is it a typo or a french variant of word debugger? 
There is more than one such typo in this review, please consider to use english everywhere.

File: src/DebuggerActions/DoesNotUnderstandDebugAction.class.st [hunk 2]
```diff
@@ -79,6 +74,11 @@ DoesNotUnderstandDebugAction >> executeAction [
 	self debugger selectTopContext
 ]
 
+{ #category : #accessing }
+DoesNotUnderstandDebugAction >> help [
+    ^ 'Create the missing method in the user prompted class and restart the debugger at that location where it can be edited.'
```
Comment (iNode) [inline src/DebuggerActions/DoesNotUnderstandDebugAction.class.st:79]: Will be such long tooltips readable in all cases even on a small displays?

File: src/DebuggerActions/ReturnValueDebugAction.class.st [hunk 3]
```diff
@@ -39,6 +39,11 @@ ReturnValueDebugAction >> executeAction [
 	self session returnValue: value from: self currentContext
 ]
 
+{ #category : #accessing }
+ReturnValueDebugAction >> help [
+	^ 'Forces a return of a given value to the previous context.'
```
Comment (iNode) [inline src/DebuggerActions/ReturnValueDebugAction.class.st]: "Force return..." or just "Return..." maybe? Just to be consistent in the form of versbs.

File: src/GT-BytecodeDebugger/GTGoAndInspectBytecodeDebugAction.class.st [hunk 4]
```diff
@@ -35,6 +35,11 @@ GTGoAndInspectBytecodeDebugAction >> executeAction [
 
 ]
 
+{ #category : #accessing }
+GTGoAndInspectBytecodeDebugAction >> help [
+	^ 'Inspects the currently selected bytecode.'
```
Comment (iNode) [inline src/GT-BytecodeDebugger/GTGoAndInspectBytecodeDebugAction.class.st]: "Inspect ..." maybe?

File: src/GT-Debugger/GTInspectSelectionDebugAction.class.st [hunk 5]
```diff
@@ -59,6 +59,11 @@ GTInspectSelectionDebugAction >> executeAction [
 
 ]
 
+{ #category : #accessing }
+GTInspectSelectionDebugAction >> help [
+	^ 'Inspecte the result of a selected piece of code from the code pane of the debugger.'
```
Comment (iNode) [inline src/GT-Debugger/GTInspectSelectionDebugAction.class.st]: "Inspecte" is it a typo as well?

File: src/GT-SUnitDebugger/GTSUnitDebuggerJumpToTestAction.class.st [hunk 6]
```diff
@@ -41,13 +41,18 @@ GTSUnitDebuggerJumpToTestAction >> executeAction [
 	self debugger stackPresentation selection: self debugger session testContext
 ]
 
+{ #category : #accessing }
+GTSUnitDebuggerJumpToTestAction >> help [
+	^ 'Select in the stack the context containing the test  method.'
```
Comment (iNode) [inline src/GT-SUnitDebugger/GTSUnitDebuggerJumpToTestAction.class.st]: Some excessive spaces.

**Review by iNode — COMMENTED** [review]
Some minor typos.  But overall looks good. Thanks for the change.

File: src/DebuggerActions/DoesNotUnderstandDebugAction.class.st (hunk 2 above)
Comment (jecisc) [reply src/DebuggerActions/DoesNotUnderstandDebugAction.class.st:79]: This is the same tooltip as before and I found it readable on my laptop. I think this length is ok.

File: src/DebuggerActions/FullStackDebugAction.class.st (hunk 1 above)
Comment (jecisc) [reply src/DebuggerActions/FullStackDebugAction.class.st]: Yes sorry! I'll push corrections. Thank you for the review.

**Review by iNode — APPROVED** [review]
Looks good to me.

**Comment by Ducasse** [comment]
I like the idea to have the help as subclassResponsibility :) like that other actions will have to have an help!!!

**Review by MarcusDenker — APPROVED** [review]
I downloaded the image that the CI build and it works (minus the bug that sometimes the help does not appear, but there is another issue for that)

**Pull request merged**